ytarchive.py

```diff
@@ -2116,29 +2120,31 @@ def main():
         "-i", new_afile
     ]
 
-    if thumbnail:
+    if thumbnail and not mergeMKV:
         ffmpeg_args.extend(["-i", new_thmbnail])
 
     if aonly:
         mfile = os.path.join(fdir, "{0}.m4a".format(fname))
     else:
-        mfile = os.path.join(fdir, "{0}.mp4".format(fname))
+        mfile = os.path.join(fdir, "{0}.{1}".format(fname, "mkv" if mergeMKV else "mp4"))
 
         ffmpeg_args.extend([
             "-i", new_vfile,
             "-movflags", "faststart"
         ])
 
-        if thumbnail:
+        if thumbnail and not mergeMKV:
             ffmpeg_args.extend([
                 "-map", "0",
                 "-map", "1",
                 "-map", "2"
             ])
 
     ffmpeg_args.extend(["-c", "copy"])
-    if thumbnail:
+    if thumbnail and not mergeMKV:
         ffmpeg_args.extend(["-disposition:v:0", "attached_pic"])
+    if thumbnail and mergeMKV:
+        ffmpeg_args.extend(["-attach", new_thmbnail,"-metadata:s:t", "mimetype=image/jpeg"])
```

Comment:
Seems the filename needs to be set as well, per https://matroska.org/technical/attachments.html#cover-art
Please add `"-metadata:s:t", "cover_land.jpg"` to the args as well.

Reply:
solved in https://github.com/Kethsar/ytarchive/pull/42/commits/981c05aa5e6c902d3532476f02d5c6b27099f1ca